plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx

```diff
@@ -96,26 +96,34 @@ export function getDefaultValueOfInheritQueryFromTemplate() {
 }
 
 /**
- * Add Product Collection block to the parent array of the Core Pagination block.
+ * Add Product Collection block to the parent or ancestor array of the Core Pagination block.
  * This enhancement allows the Core Pagination block to be available for the Product Collection block.
  */
-export const addProductCollectionBlockToParentOfPaginationBlock = () => {
+export const addProductCollectionToQueryPaginationParentOrAncestor = () => {
 	if ( isWpVersion( '6.1', '>=' ) ) {
 		addFilter(
 			'blocks.registerBlockType',
 			'woocommerce/add-product-collection-block-to-parent-array-of-pagination-block',
 			( blockSettings: Block, blockName: string ) => {
-				if (
-					blockName !== coreQueryPaginationBlockName ||
-					! blockSettings?.parent
-				) {
+				if ( blockName !== coreQueryPaginationBlockName ) {
 					return blockSettings;
 				}
 
-				return {
-					...blockSettings,
-					parent: [ ...blockSettings.parent, blockJson.name ],
-				};
+				if ( blockSettings?.ancestor ) {
+					return {
+						...blockSettings,
+						ancestor: [ ...blockSettings.ancestor, blockJson.name ],
+					};
+				}
+
+				if ( blockSettings?.parent ) {
+					return {
+						...blockSettings,
+						parent: [ ...blockSettings.parent, blockJson.name ],
+					};
+				}
```

Comment:
Nit: Maybe we can add a comment here that this is done to support older versions of Wordpress. 

Also, for now, we [requires at least: 6.4](https://github.com/woocommerce/woocommerce/blob/677aef3adf632315ede1ef88a441139c3a392503/plugins/woocommerce/woocommerce.php#L11), but when we update this version to 6.5, we may consider removing these lines of code. no?

Reply:
Sure, added in https://github.com/woocommerce/woocommerce/pull/47749/commits/b6e98dcc115f01ac1400b6989b8f6a595eb94608